parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -47,21 +48,33 @@
 import org.apache.parquet.schema.MessageType;
 
 import static org.apache.iceberg.parquet.ParquetConversions.fromParquetPrimitive;
+import static org.apache.iceberg.util.BinaryUtil.truncateBinaryMax;
+import static org.apache.iceberg.util.BinaryUtil.truncateBinaryMin;
+import static org.apache.iceberg.util.UnicodeUtil.truncateStringMax;
+import static org.apache.iceberg.util.UnicodeUtil.truncateStringMin;
 
 public class ParquetUtil {
   // not meant to be instantiated
   private ParquetUtil() {
   }
 
   public static Metrics fileMetrics(InputFile file) {
+    return fileMetrics(file, TableProperties.WRITE_METADATA_TRUNCATE_BYTES_DEFAULT);
+  }
+
+  public static Metrics fileMetrics(InputFile file, int statsTruncateLength) {
     try (ParquetFileReader reader = ParquetFileReader.open(ParquetIO.file(file))) {
-      return footerMetrics(reader.getFooter());
+      return footerMetrics(reader.getFooter(), statsTruncateLength);
     } catch (IOException e) {
       throw new RuntimeIOException(e, "Failed to read footer of file: %s", file);
     }
   }
 
   public static Metrics footerMetrics(ParquetMetadata metadata) {
+    return footerMetrics(metadata, TableProperties.WRITE_METADATA_TRUNCATE_BYTES_DEFAULT);
```

Comment:
I don't think there is a need for this to be left. The truncate length should always be included when getting metrics.

The version that uses the default is called from two places:
* `ParquetWriteAdapter` that is created by `Parquet.write` when the internal `ParquetWriter` is not used
* `SparkTableUtil` that reads Parquet footers when converting Hive tables to Iceberg metadata

I think both should be updated. The write adapter should use the config setting, and the Spark util method can default this inline. Then we can get rid of this method.

Reply:
Yes, would be good to get rid of this. I updated the PR with these changes.